GetMessages/handler.ts

```diff
@@ -62,17 +70,55 @@ type IGetMessagesHandler = (
 export function GetMessagesHandler(
   messageModel: MessageModel
 ): IGetMessagesHandler {
-  return async fiscalCode => {
+  return async (
+    fiscalCode,
+    maybePageSize,
+    maybeEnrichResultData,
+    maybeMaximumId: O.Option<NonEmptyString>,
+    maybeMinimumId: O.Option<NonEmptyString>
+  ) => {
+    const pageSize = pipe(
+      maybePageSize,
+      O.getOrElse(() => defaultPageSize)
+    );
+    // TODO: Consider enrichResultData to enrich messages with content
+    // tslint:disable-next-line:no-unused-variable no-dead-store
+    const enrichResultData = pipe(
+      maybeEnrichResultData,
+      O.getOrElse(() => false)
+    );
     return pipe(
-      messageModel.findMessages(fiscalCode),
+      TE.Do,
+      TE.bind("maximumId", () =>
+        pipe(
+          maybeMaximumId,
+          O.getOrElse(() => undefined),
+          TE.of
+        )
+      ),
+      TE.bind("minimumId", () =>
+        pipe(
+          maybeMinimumId,
+          O.getOrElse(() => undefined),
+          TE.of
+        )
+      ),
```

Comment:
A more compact version :)
```suggestion
      TE.bind("maximumId", () => TE.of(O.toUndefined(maybeMaximumId))),
      TE.bind("minimumId", () => TE.of(O.toUndefined(maybeMinimumId))),
```